src/interactivity/win32/windowio.cpp

```diff
@@ -78,7 +78,13 @@ VOID SetConsoleWindowOwner(const HWND hwnd, _Inout_opt_ ConsoleProcessHandle* pP
     else
     {
         // Find a process to own the console window. If there are none then let's use conhost's.
-        pProcessData = gci.ProcessHandleList.GetFirstProcess();
+        pProcessData = gci.ProcessHandleList.FindProcessInList(ConsoleProcessList::ROOT_PROCESS_ID);

```

Comment:
This line feels like a scary change that might stealth break something we can't immagine

Reply:
So, I went through all the callers of this function. They either provide the root process, OR there is no root process... at which point, a random process is selected. The oldest one still connected.

I think it's safe to say "the root process IS the oldest, and we should default to that if we can. Otherwise, we can choose the NEXT oldest"

Reply:
Ah btw, `GetFirstProcess()` returns the newest process and not the oldest. See: https://github.com/microsoft/terminal/pull/14421#discussion_r1028609903